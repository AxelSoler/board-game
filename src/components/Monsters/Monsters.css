.grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 90%;
  margin: 48px auto;
}

.monsterInput {
  width: 50%;
  padding: 12px;
  font-size: 18px;
}
