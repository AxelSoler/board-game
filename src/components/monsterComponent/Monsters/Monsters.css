.grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 90%;
  height: 59.4vh;
  margin: 48px auto;
  overflow-y: scroll;
}

.monsterInput {
  width: 50%;
  padding: 12px;
  font-size: 18px;
}
