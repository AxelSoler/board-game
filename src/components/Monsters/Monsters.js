import React from 'react';
import { useSelector } from 'react-redux';
import { getListOfMonsters } from '../../redux/monsters/monsters';
import store from '../../redux/configureStore';
import Monster from '../Monster/Monster';
import './Monsters.css';

const Monsters = () => {
  const allMonsters = useSelector((state) => state);
  if (allMonsters.length === 0) {
    store.dispatch(getListOfMonsters());
  }

  return (
    <div className="grid">
      {allMonsters.map((monster) => (
        <Monster key={monster.monsterId} id={monster.monsterId} name={monster.monsterName} />
      ))}
    </div>
  );
};

export default Monsters;
