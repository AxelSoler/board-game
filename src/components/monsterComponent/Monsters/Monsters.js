import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { getListOfMonsters } from '../../../redux/monsters/monsters';
import store from '../../../redux/configureStore';
import Monster from '../Monster/Monster';
import Header from '../../Header/Header';
import './Monsters.css';

const Monsters = () => {
  const [Filter, setFilter] = useState('');
  const allMonsters = useSelector((state) => state.allMonsters);
  if (allMonsters.length === 0) {
    store.dispatch(getListOfMonsters());
  }
  const onChange = (e) => {
    const newFilter = e.target.value;
    setFilter(newFilter);
  };

  return (
    <div>
      <Header name="MONSTERS" />
      <div className="team-list-header">
        <p>SEARCH</p>
        <input className="monsterInput" onChange={onChange} type="text" placeholder="Monster..." />
      </div>
      <div className="grid">
        {allMonsters.filter((item) => {
          if (Filter === '') {
            return item;
          } if (item.monsterName.toLowerCase().includes(Filter.toLowerCase())) {
            return item;
          }
          return false;
        }).map((monster) => (
          <Monster key={monster.monsterId} id={monster.monsterId} name={monster.monsterName} />
        ))}
      </div>
    </div>
  );
};

export default Monsters;
