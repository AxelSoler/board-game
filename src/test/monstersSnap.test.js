import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Monsters from '../components/Monsters/Monsters';

test('monsters snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Monsters />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
