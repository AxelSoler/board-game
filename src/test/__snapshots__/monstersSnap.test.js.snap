// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`monsters snapshot 1`] = `
<div>
  <div
    className="headerContainer"
  >
    <a
      aria-current="page"
      className="backBtn active"
      href="/"
      onClick={[Function]}
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="2.5em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#fff",
          }
        }
        viewBox="0 0 512 512"
        width="2.5em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
        />
      </svg>
    </a>
    <header>
      <h3>
        MONSTERS
      </h3>
    </header>
  </div>
  <div
    className="team-list-header"
  >
    <p>
      SEARCH
    </p>
    <input
      className="monsterInput"
      onChange={[Function]}
      placeholder="Monster..."
      type="text"
    />
  </div>
  <div
    className="grid"
  />
</div>
`;
